source/server-selection/tests/README.rst

```diff
@@ -65,3 +65,43 @@ the TopologyDescription. Each YAML file contains a key for these stages of serve
 Drivers implementing server selection MUST test that their implementation
 correctly returns the set of servers in ``in_latency_window``. Drivers SHOULD also test
 against ``suitable_servers`` if possible.
+
+Selection Within Latency Window Tests
+>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
+
+Each YAML file for these tests has the following format:
+
+- ``topology_description``: the state of a mocked cluster
+
+- ``in_window``: array of servers in the latency window that the selected server
+  is to be chosen from. Each element will have all of the following fields:
+
+  - ``address``: a unique address identifying this server
+
+  - ``active_connection_count``: the number of active connections this server
+    currently has open
+
+  - ``available_connection_count``: the number of available connections this
+    server currently has in its pool.
+
+  - ``wait_queue_length``: the number of requests in the wait queue
+    for this server's pool.
+
+- ``max_pool_size``: the maximum number of connections allowed in a server's
+  connection pool.
+
+- ``expected_frequencies``: a document whose keys are the server addresses from the
+  ``in_window`` array and values are numbers in [0, 1] indicating the frequency
+  at which the server should have been selected.
+
+For each file, pass the information from `in_window` to whatever function is
```

Comment:
The test should also configure the read preference used for server selection. I suggest adding a `read_preference` field to be consistent with the existing tests.

Reply:
I think we discussed this in the meeting, but since this test is verifying what happens after the set of available servers is determined and the topologies will always be sharded, a read preference shouldn't be necessary here.

Reply:
Ah I see, yeah that makes sense. In my implementation I skipped right to the in window portion, so I didn't need a TopologyDescription or a read preference. I imagine most drivers will probably implement your way though. 

Rather than including a read preference in each spec test, I just updated the runner's description to say use a default or "primary" read preference since it shouldn't matter for sharded topologies.

Reply:
It still feels a little odd that we have no explicit tests for replica sets. Perhaps the prose test can cover it?